Dolphin scripts/Entrance Randomizer/lib/graph_creation.py

```diff
@@ -92,31 +109,79 @@ def create_vertices(
     return output_text
 
 
-def create_edges(transitions_map: Mapping[tuple[int, int], tuple[int, int]]):
-    connections = [(original[0], redirect[1]) for original, redirect in transitions_map.items()]
-    connections_two_way: list[tuple[int, int]] = []
-    connections_one_way: list[tuple[int, int]] = []
+def edge_component(
+    start: int,
+    end: int,
+    counter: int,
+    direct: Direction,
+    color: str | None,
+    line_type: LineType,
+):
+    direct_str = str(direct == Direction.ONEWAY).lower()
+    output = (
+        f'<edge source="{transition_infos_dict_rando[start].area_id}" '
+        + f'target="{transition_infos_dict_rando[end].area_id}" '
+        + f'isDirect="{direct_str}" '
+        + f'id="{counter}"'
+    )
+    if line_type == LineType.DASHED or color is not None:
+        output += create_own_style({
+            "strokeStyle": color,
+            "lineDash": "2" if line_type == LineType.DASHED else None,
+        })
```

Comment:
No it would not give the desired result if this is called with all values being None.
Yes I could rewrite the function to return an empty string if all values are None, but aren't we just moving the problem at that point?
Unless you can tell me how to cleanly do that check at the start of the function...

Reply:
To avoid having to iterate on all values twice, you could store the result of `",".join` in a variable and check if it's an empty string.
If all values are empty, `",".join([... if value is not None])` is equivalent to `",".join([])` which evaluates to `""`

The implementation can look something like this:
```py
def create_own_style(params: dict[str, str | None]):
    output_style = ",".join([
        f"&quot;{key}&quot;:&quot;{value}&quot;"
        for key, value in params.items()
        if value is not None
    ])
    return ' ownStyles="{&quot;0&quot;:{' + output_style + '}}"' if output_style else ""
```

Reply:
Yeah that looks good, added!